recipe/meta.yaml

```diff
@@ -1,5 +1,5 @@
 {% set version = "3.11.0" %}
-{% set build = 2 %}
+{% set build = 3 %}
```

Comment:
Probably not needed b/c at the time of the previous build there was no cython 4. The metadata fix is desirable though.

Reply:
I don't understand this.  I would not expect the cython version used for the build to make it into the hash so we need to bump build number to avoid any conflicts?

Reply:
> I don't understand this. I would not expect the cython version used for the build to make it into the hash so we need to bump build number to avoid any conflicts?

The change here goes from `cython >=0.29.31,<1` to `cython >=0.29.31,<4` in the build/host dependencies. That means, unless you do want a new build with `cython>1,<4`, the current build is OK according to the package metadata. There won't be any conflicts b/c cython is used only to build, unless I'm missing something obvious here.

Reply:
I don't know enough about how cython works to say for sure whether the builds with `cython <1` will be exactly equivalent to builds with `cython >=1` that this PR enables. I'd hope they're just as good and functional?

Reply:
we would want to publish this new build though, I'd think? to find out if it's working fine for users downstream? the more consequential rebuilds will happen when the bot opens a python 3.13 migration PR, which I think this PR can unblock whether or not the new higher-cython-version builds get published for `python <=3.12`.

Reply:
If the current builds are working there is no need to re-build.  Cython generates c-code which are then compiled as the extension modules.  To go fast cython uses some semi-private / unstable c-APIs to CPython so when you go across CPython minor versions it is frequently (always?) required to also move to a newer cython that has adapted to the upstream changes.  However, once the c-code is generated and it complies, then there is no difference (maybe performance, but I think that can go both ways) on which version of cython was used.

Or put another way, if users can tell which version of cython we used that is a cython bug!

100% agree this is needed to un-block the 3.13 migration.

Reply:
Gotcha, I don't have strong feelings either way, can absolutely defer to maintainers' preference for how to tweak this PR. Are you proposing dropping the build number back to 2 and (squash?) merging with [ci skip]?

I'm not personally making use of h5py in anything right now, just saw on the status page that it was blocking py313 migration for a lot of downstream packages and I made basically the same cython bound change to ruamel.yaml.clib which is in the stack of one of the packages I maintain

Reply:
I'm going to also defer to @ocefpaf ;)

Reply:
The penalty for a new build number is storage. It is not a big deal, while unnecessary IMO, we can merge as is.